### DISCUSSION TITLE
After upgrading to v2, the built binary package is not usable.

### DISCUSSION THREAD

**Discussion by kerwin612** [tauri-apps/tauri#11656] [Q&A]
The project source code is here: https://github.com/kerwin612/PortableRunner/tree/react  
> Note: The following feedback is based on branch **react** (which is based on tauri v2); the main branch is ok (based on tauri v1) . 
-----  
**The binary package output during the build runs abnormally:** 
* `pnpm tauri build`
The main page of the program is abnormal and fails to load: 
![image](https://github.com/user-attachments/assets/609d1db1-9585-4deb-a2f8-1359a9ee2e9c)
The right-click menu does not appear on the system tray: 
![image](https://github.com/user-attachments/assets/db83b265-e117-4978-be87-b2d9e7fe3021)
The screenshot of the Task Manager is as: 
![image](https://github.com/user-attachments/assets/033c294f-25d7-494e-a566-d3317ea44060)
-----  
**During debugging, everything is okay:** 
* `pnpm tauri dev`
![image](https://github.com/user-attachments/assets/b8d8987b-f948-470c-a551-4a68db697352)
![image](https://github.com/user-attachments/assets/8370f467-5ed8-4d35-b818-dfecd07a4a93)
![image](https://github.com/user-attachments/assets/8e8fe5f6-2903-4200-a8c3-97df28580527)

**Comment by FabianLars** [comment]
For issues like this, always try a debug build `pnpm tauri build --debug` (output in target/debug/). If you open that exe from inside a terminal you may get helpful output. In this case i get this:
```
thread 'main' panicked at src/main.rs:67:38:
called `Result::unwrap()` on an `Err` value: Custom { kind: NotFound, error: "invalid HOME" }
```
which i don't get in dev. could that be it? (the line it reports doesn't make sense to me tho)

**Reply by kerwin612** [reply]
Thank you for your reply. After testing with your method, I found that it might be caused by file permissions. Thanks again!

**Reply by kerwin612** [reply]
![image](https://github.com/user-attachments/assets/e67c00cb-d140-4f60-8834-2b07e83ce3c1)

It's strange that this white screen still occasionally appears, and there are no logs in the console. How should I troubleshoot this situation?

**Reply by FabianLars** [reply]
If the titlebar still works then this means the webview crashed. Could you try commenting out all of the tauri commands on the rust side and see if it works better then? Wondering if there's idk a deadlock or something. Maybe same for the setup hook.

What confuses me a bit is that it only happens occasionally so maybe a race condition.

**Reply by kerwin612** [reply]
After careful testing, in the situation shown in the screenshot above:

① Pressing F12 does not open Edge's debugging window.
② The system-tray is also abnormal, tooltips do not appear, the right-click menu does not show up, and clicking does not trigger the display or hiding of windows.

**Reply by FabianLars** [reply]
if the tray is also broken i think it most likely crashes in the setup hook but that should always show something in the console...

**Reply by kerwin612** [reply]
Yes, the most crucial thing is that there are no abnormal logs to be seen; and everything is OK during `tauri dev`, so it's really confusing and I don't know where to resolve.

**Reply by kerwin612** [reply]
Still based on the code of this branch: https://github.com/kerwin612/PortableRunner/tree/react
### I made the following changes:
I changed **main.rs** to the following content:
```rs
// Prevents additional console window on Windows in release, DO NOT REMOVE!!
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]

fn main() {
    tauri::Builder::default()
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```
Then, I deleted the following content from the **tauri.conf.json** file:
```txt
-    "beforeDevCommand": "pnpm dev",
-    "devUrl": "http://localhost:1420",
-    "beforeBuildCommand": "pnpm build",
```
Then, I deleted all files under the dist directory except for **index.html**;   
Finally, I changed **index.html** to the following content:
```html
<!doctype html>
<html lang="en">
  <body>test</body>
</html>
```

Finally, I re-executed `pnpm tauri build --debug`, but the result was still the same as the screenshot feedback before, with no changes.

**Reply by kerwin612** [reply]
I used `pnpm create tauri-app` to recreate a new project, then copied my business logic code into the new project, and then ran `pnpm tauri build` to build it. Surprisingly, the issue was gone.

I am not sure what caused the problem. You can compare the two versions of the code to see if you can figure out what caused the issue.

Code version with the issue: https://github.com/kerwin612/PortableRunner/tree/829dec792c44818d009f2f4b67c79af20510ce61

Code version without the issue: https://github.com/kerwin612/PortableRunner/tree/cfd8ca34781ea20e1360bf70a8b1adadecf09c69

Code comparison: https://github.com/kerwin612/PortableRunner/commit/cfd8ca34781ea20e1360bf70a8b1adadecf09c69

**Reply by FabianLars** [reply]
hmm, i'm glad that it works for you now but i don't see any relevant changes there 🤔 

The global-shortcut rust package was removed but that shouldn't be able to crash the webview. The removed js packages weren't even imported so they couldn't have been the cause either.

Maybe it's one of the Cargo.lock updates but really, your guess is as good as mine.

Thanks for the links btw
